unit_py/test_double_spring_mass.py

```diff
@@ -179,5 +179,122 @@ def external_force(t, q, v, f):
+        # mass is the opposite of that of the first mass to provide a constant
+        # output position.
+        self.A[3, :] = -self.A[2, :]
+        x_analytical = np.stack([expm(self.A * t) @ self.x0 for t in time], axis=0)
+
+        # Compare the numerical and analytical solutions
+        self.assertTrue(np.allclose(x_jiminy, x_analytical, atol=TOLERANCE))
+
+    def test_freeflyer_multiple_constraints(self):
+        '''
+        @brief Test having several constraints at once.
+        @details This test features:
+                     - a freeflyer with a fixed body constraint on the freeflyer
+                    (this gives a non-trivial constraint to solve to effectively cancel the freeflyer)
+                     - a fixed body constaint on the output mass.
+        '''
+        # Rebuild the model with a freeflyer.
+        self.robot = jiminy.Robot()
+        self.robot.initialize("data/linear_two_masses.urdf", has_freeflyer=True)
+        for joint_name in ["FirstJoint", "SecondJoint"]:
+            motor = jiminy.SimpleMotor(joint_name)
+            self.robot.attach_motor(motor)
+            motor.initialize(joint_name)
+
+        # Configure robot
+        model_options = self.robot.get_model_options()
+        motor_options = self.robot.get_motors_options()
+        model_options["joints"]["enablePositionLimit"] = False
+        model_options["joints"]["enableVelocityLimit"] = False
+        for m in motor_options:
+            motor_options[m]['enableTorqueLimit'] = False
+            motor_options[m]['enableRotorInertia'] = False
+        self.robot.set_model_options(model_options)
+        self.robot.set_motors_options(motor_options)
+
+        # Set same spings as usual
+        def compute_command(t, q, v, sensor_data, u):
+            u[:] = 0.0
+
+        def internal_dynamics(t, q, v, sensor_data, u):
+            u[6:] = - self.k * q[7:] - self.nu * v[6:]
+
+        controller = jiminy.ControllerFunctor(compute_command, internal_dynamics)
+        controller.initialize(self.robot)
+        engine = jiminy.Engine()
+        engine.initialize(self.robot, controller)
+        # Disable gravity.
+        engine_options = engine.get_options()
+        engine_options["world"]["gravity"] = np.zeros(6) # Turn off gravity
+        engine.set_options(engine_options)
+
+        # Add a kinematic constraints.
+        freeflyer_constraint = jiminy.FixedFrameConstraint("world")
+        self.robot.add_constraint("world", freeflyer_constraint)
+        fix_mass_constraint = jiminy.FixedFrameConstraint("SecondMass")
+        self.robot.add_constraint("fixMass", fix_mass_constraint)
+
+        # Initialize with zero freeflyer velocity...
+        x_init = np.zeros(17)
+        x_init[7:9] = self.x0[:2]
+        x_init[-2:] = self.x0[2:]
+        # ... and a "random" (but fixed) freeflyer quaternion
+        x_init[:7] = np.array([0.5, -0.8, 42.0, 0.1, 0.5, -0.7, 0.1])
```

Comment:
I don't like const random. I would rather reset the random generator of numpy, set the root, and asked it to generate it. Unless your values are not this much random x)